src/Http/UrlGenerator.php

```diff
@@ -59,4 +66,19 @@ public function to($collection)
     {
         return $this->routes[$collection];
     }
+
+    /**
+     * Generate a URL to an instance of a resource.
+     *
+     * @param string $resourceClass
+     * @param AbstractModel $instance
+     * @param $args
+     * @return void
+     */
+    public function toResource(string $resourceClass, AbstractModel $instance, ...$args): string
```

Comment:
Do we want to explicitly pass in the resource type, or extrapolate via calling `get_class()` on the instance? `get_class` would be cleaner on invokation, but then we'd need to check all parent classes

Reply:
Given that passing in the class type we expect causes PHP itself to go through the classes parents and stuff automatically in C, I don't see how using `get_class()` would be beneficial, especially since it would probably require more code than it's worth (slowing things down a bit)